### PR TITLE
Doc: Improve Mesh sensor documentation

### PR DESCRIPTION
**Pull request by ludvigsj**
This adds snippets showing how to initialize sensor settings and descriptors, as well as clarifying some potentially confusing points about sensor types.

Signed-off-by: Ludvig Samuelsen Jordet <ludvig.jordet@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/libraries/bluetooth_services/mesh/sensor.rst [hunk 1]
```diff
@@ -214,6 +214,26 @@ A sensor's Descriptor contains parameters that may aid other mesh nodes in inter
...
+           .negative = { .val1 = 0, .val2 = 750000 },
+           .positive = { .val1 = 3, .val2 = 500000 },
+       },
+       .sampling_type = BT_MESH_SENSOR_SAMPLING_ARITHMETIC_MEAN,
+       .period = 300,
+       .update_interval = 50
+   };
+
+   struct bt_mesh_sensor temp_sensor = {
+       .type = &bt_mesh_sensor_present_dev_op_temp,
```
Comment (Balaklaka) [inline doc/nrf/libraries/bluetooth_services/mesh/sensor.rst:232]: We have a description of the descriptor, but not type. Should we add "bt_mesh_sensor_present_dev_op_temp" too? Or is it to much information, as we maybe have to do this for all 3 places where type is used.

File: doc/nrf/libraries/bluetooth_services/mesh/sensor.rst [hunk 2]
```diff
@@ -333,6 +353,50 @@ Each entry in the list has a type and two access callbacks, and the list should
...
+                                   struct bt_mesh_msg_ctx *ctx,
+                                   const struct sensor_value *value)
+   {
+        /** Store incoming threshold in application-defined way.
+         *  Return error code to reject set.
+         */
+        return set_threshold(value);
+   }
+
+   static struct bt_mesh_sensor_setting settings[] = {
```
Comment (Balaklaka) [inline doc/nrf/libraries/bluetooth_services/mesh/sensor.rst:384]: Can this be const? And then remove the typecast on line 396.

File: doc/nrf/libraries/bluetooth_services/mesh/sensor.rst (hunk 1 above)
Comment (ludvigsj) [reply doc/nrf/libraries/bluetooth_services/mesh/sensor.rst:232]: The type(s) are described in the sensor_types.rst document. What do you suggest to add here?

File: doc/nrf/libraries/bluetooth_services/mesh/sensor.rst (hunk 2 above)
Comment (ludvigsj) [reply doc/nrf/libraries/bluetooth_services/mesh/sensor.rst:384]: This was a lazy code-grap from the sensor_server sample application on my part, the typecast is not needed even if this is not const, the array can be passed bare and will degrade to a (const) pointer just fine. Will update with more readable code.

**Review by Balaklaka — APPROVED** [review]

**Review by mia-ko — CHANGES_REQUESTED** [review]
To me it looks like we're using wrong heading types for sections in sensor.rst, h2 and h3 seem to be switched around. At least according to this rule:
Document title (h1) use “#” for the underline character
First section heading level (h2) use “*”
Second section heading level (h3) use “=”

**Review by mia-ko — APPROVED** [review]

**Pull request merged**